### ISSUE TITLE
Is an undecimated DWT possible?

### ISSUE THREAD

**Issue by bryaan** [JuliaDSP/Wavelets.jl#31]
Is there any way to create an undecimated DWT as of now?

If not, are there any plans?

**Comment by gummif** [comment]
No, the upsampling and downsampling is pretty baked into the algorithms. I at least don't have much time so there are no plans.

**Comment by daanhb** [comment]
Perhaps I can help? It is not that I have much time available right now myself, but some time ago I experimented with Wavelets.jl in order to play with vectors of arbitrary length, biorthogonal (filter-based) transforms and various extension methods. This didn't quite materialize yet, but I intend to pick it up again in the not too distant future. If so, I'd be happy to polish and contribute the results.

**Comment by ElOceanografo** [comment]
I've just implemented some pseudocode from Percival and Walden's "Wavelet methods for time series analysis" for the undecimated DWT (which they call the maximal overlap DWT, or MODWT).  I've checked my code's results against those of the `wavMODWT` function in the R package `wmtsa`, based on the book.  I can reproduce them, though it requires some fiddling with wavelet filters...theirs seem to be defined slightly differently from those here (differences in sign, normalization, and indexing of the filter orders).  

I could add this implementation and make a PR if desired...
